### DISCUSSION TITLE
process is not happening

### DISCUSSION THREAD

**Discussion by afishler-zz** [OptimalBits/bull#2343] [Q&A]
Must be a totally silly issue but I am really a newbie.  Process does not seem to happen at all. The debug message is not printed at all - only the "Main start". After that it just hangs.

After a while I get
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "undefined".] {
  code: 'ERR_UNHANDLED_REJECTION'
}

The code

```ts
import Queue from "bull";

const queue = new Queue("myQueue");

const main = async () => {
  console.log("Main start");
  await queue.add({ name: "John", age: 30 });
};

queue.process((job, done) => {
  console.log(job.data);
  done();
});

main().catch(console.error);
```

**Comment by manast** [comment]
I quickly tested with this code and it just works:
```ts
const Queue = require ("bull");

const queue = new Queue("myQueue");

const main = async () => {
  console.log("Main start");
  await queue.add({ name: "John", age: 30 });
};

queue.process((job, done) => {
  console.log(job.data);
  done();
});

main().catch(console.error);
```

<img width="213" alt="image" src="https://user-images.githubusercontent.com/95200/164126124-4b1843a1-7570-4a48-a525-718469d3fd5e.png">

**Reply by afishler-zz** [reply]
Thanks you. Just realized Redis needs to run as an independent server. I thought it was embedded in Bull somehow
